izanami-server/app/store/dynamo/DynamoJsonDataStore.scala

```diff
@@ -0,0 +1,198 @@
+    val request = new PutItemRequest()
+      .withTableName(tableName)
+      .withConditionExpression(if (createOnly) "attribute_not_exists(id)" else null)
+      .withItem(
+        Map(
+          "id"    -> new AttributeValue().withS(id.key),
+          "store" -> new AttributeValue().withS(storeName),
+          "value" -> DynamoMapper.fromJsValue(data)
+        ).asJava
+      )
+
+    client
+      .single(request)
+      .map(_ => Result.ok(data))
+      .toF
+      .recover {
+        case _: ConditionalCheckFailedException => Result.errors[JsValue](ErrorMessage("error.data.exists", id.key))
+      }
+  }
+
+  override def delete(id: Key): F[Result[JsValue]] =
+    deleteF(id).toF
+
+  def deleteF(id: Key): Future[Result[JsValue]] = {
+    Logger.debug(s"Dynamo delete on $tableName and store $storeName delete with id : $id")
+
+    val request = new DeleteItemRequest()
+      .withTableName(tableName)
+      .withConditionExpression("attribute_exists(id)")
+      .withReturnValues(ReturnValue.ALL_OLD)
+      .withKey(
+        Map(
+          "id"    -> new AttributeValue().withS(id.key),
+          "store" -> new AttributeValue().withS(storeName)
+        ).asJava
+      )
+
+    client
+      .single(request)
+      .map(_.getAttributes.get("value"))
+      .map(DynamoMapper.toJsValue)
+      .map(Result.ok)
+      .recover { case _: ConditionalCheckFailedException => Result.error[JsValue](s"error.data.missing") }
+  }
+
+  override def deleteAll(patterns: Seq[String]): F[Result[Done]] = {
+    Logger.debug(s"Dynamo query on $tableName and store $storeName deleteAll with patterns : $patterns")
+
+    getByIdLike(patterns)
+      .mapAsync(10) { case (id, _) => deleteF(id) }
+      .runWith(Sink.ignore)
+      .toF
+      .map(_ => Result.ok(Done.done()))
+  }
+
+  override def getByIdLike(patterns: Seq[String]): Source[(Key, JsValue), NotUsed] = {
+    Logger.debug(s"Dynamo query on $tableName and store $storeName getByIdLike with patterns : $patterns")
+    findKeys(patterns)
+  }
+
+  override def getById(id: Key): F[Option[JsValue]] = {
+    Logger.debug(s"Dynamo query on $tableName and store $storeName getById with id : $id")
+
+    val request = new GetItemRequest()
+      .withTableName(tableName)
+      .withKey(
+        Map(
+          "id"    -> new AttributeValue().withS(id.key),
+          "store" -> new AttributeValue().withS(storeName)
+        ).asJava
+      )
+
+    client
+      .single(request)
+      .toF
+      .map(r => Option(r.getItem).map(_.get("value").getM).map(DynamoMapper.toJsValue))
+  }
+
+  override def getByIdLike(patterns: Seq[String], page: Int, nbElementPerPage: Int): F[PagingResult[JsValue]] = {
+    Logger.debug(
+      s"Dynamo query on $tableName and store $storeName getByIdLike with patterns : $patterns, page $page, $nbElementPerPage elements by page"
+    )
+
+    val position = (page - 1) * nbElementPerPage
+    findKeys(patterns)
```

Comment:
Here, you're looking for keys and then you fetch datas by id. 
I think you could fetch and filter data in one query, it could be more performant !

Reply:
Hi, I don't think there are multiple queries, there is only a QueryRequest in `findKeys`  and here we just iterate over all the results until we find the page we are interested in

Reply:
Ok sorry, I don't know why but I thought there were a getById after ... That's fine I will merge the PR

Reply:
Ok ^^

Thanks a lot !

Reply:
Thanks a lot for your PR !